apis/managedkafka/v1alpha1/topic_types.go

```diff
@@ -0,0 +1,116 @@
+// +kcc:proto=google.cloud.managedkafka.v1.Topic
+type ManagedKafkaTopicSpec struct {
+	commonv1alpha1.CommonSpec `json:",inline"`
+
+	// +required
+	Location string `json:"location"`
+
+	// +required
+	ClusterRef *ClusterRef `json:"clusterRef"`
+
+	// The ManagedKafkaTopic name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Required. The number of partitions this topic has. The partition count can
+	//  only be increased, not decreased. Please note that if partitions are
+	//  increased for a topic that has a key, the partitioning logic or the
+	//  ordering of the messages will be affected.
+	// +kcc:proto:field=google.cloud.managedkafka.v1.Topic.partition_count
+	PartitionCount *int32 `json:"partitionCount,omitempty"`
+
+	// Required. Immutable. The number of replicas of each partition. A
+	//  replication factor of 3 is recommended for high availability.
+	// +kcc:proto:field=google.cloud.managedkafka.v1.Topic.replication_factor
+	ReplicationFactor *int32 `json:"replicationFactor,omitempty"`
+
+	// Optional. Configurations for the topic that are overridden from the cluster
+	//  defaults. The key of the map is a Kafka topic property name, for example:
+	//  `cleanup.policy`, `compression.type`.
+	// +kcc:proto:field=google.cloud.managedkafka.v1.Topic.configs
+	Configs map[string]string `json:"configs,omitempty"`
+}
+
+// ManagedKafkaTopicStatus defines the config connector machine state of ManagedKafkaTopic
+type ManagedKafkaTopicStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the ManagedKafkaTopic resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *ManagedKafkaTopicObservedState `json:"observedState,omitempty"`
+}
+
+// ManagedKafkaTopicSpec defines the desired state of ManagedKafkaTopic
+// +kcc:proto=google.cloud.managedkafka.v1.Topic
+// ManagedKafkaTopicObservedState is the state of the ManagedKafkaTopic resource as most recently observed in GCP.
+type ManagedKafkaTopicObservedState struct {
+	// Identifier. The name of the topic. The `topic` segment is used when
+	//  connecting directly to the cluster. Structured like:
+	//  projects/{project}/locations/{location}/clusters/{cluster}/topics/{topic}
+	// +kcc:proto:field=google.cloud.managedkafka.v1.Topic.name
+	Name *string `json:"name,omitempty"` // TODO(jingyih): remove this field as it is the same as externalRef
```

Comment:
Wait, any reasons to add this field (with a cleanup TODO) if this is a pure new Alpha resource?

Reply:
Yeah this is a new resource. I am removing the field in a followup PR.